File: lambda/lambda_function.py
import boto3
import os, json

bucketName = os.environ['BUCKET_NAME']

def lambda_handler(event, context):
    try:
        fileName   = event['fileName']
        objectKey  = f"image/{fileName}"
        signedURL = boto3.client('s3').generate_presigned_url(
            ClientMethod='put_object', 
            Params={'Bucket': bucketName, 'Key': objectKey},
            ExpiresIn=300)
        return {
            "statusCode": 200,
            "body": json.dumps({
                "signedURL": signedURL
            })
        }
    except KeyError as error:
        return {
            "statusCode": 400,
            "body": error
        }
    except Exception as error:
        return {
            "statusCode": 500,
            "body": error
        }

File: lambda/lambda_function.js
const AWS = require('aws-sdk')
AWS.config.update({ region: process.env.AWS_REGION })
const s3 = new AWS.S3()
const URL_EXPIRATION_SECONDS = 300

// Main Lambda entry point
exports.handler = async (event) => {
  if ('fileName' in event && 'fileType' in event) {
    return await getSignedURL(event)
  } else {
    return {
      statusCode: 400,
      body: JSON.stringify({
        error: "Bad request, have no fileName or fileType in body"
      })
    }
  }
}

const getSignedURL = async (event) => {
  const fileName = event['fileName']
  const fileType = event['fileType']
  const objectKey = `image/${fileName}`
  // Get signed URL from S3
  const s3Params = {
    Bucket: process.env.bucketName,
    Key: objectKey,
    Expires: URL_EXPIRATION_SECONDS,
    ContentType: fileType
  }
  const signedURL = await s3.getSignedUrlPromise('putObject', s3Params)
  return {
    statusCode: 200,
    body: JSON.stringify({
      signedURL: signedURL
    })
  }
}